python/pyiceberg/io/__init__.py

```diff
@@ -218,11 +234,53 @@ def delete(self, location: Union[str, InputFile, OutputFile]) -> None:
         """
 
 
-def load_file_io(_: Properties) -> FileIO:
-    # To be implemented in a different PR.
-    # - If py-file-io is present, load the right Python class
-    #   - When the property is missing, map from Java's filo-io to an appropriate FileIO
-    # - Extend the FileIO structure with a initialize that pass in properties (could also be the constructor?)
+ARROW_FILE_IO = "pyiceberg.io.pyarrow.PyArrowFileIO"
+
+# Mappings from the Java FileIO impl to a Python one. The list is ordered by preference.
+# If a implementation isn't installed, it will fall back to the next one.
+JAVA_FILE_IO_MAPPINGS: Dict[str, List[str]] = {
+    "org.apache.iceberg.dell.ecs.EcsFileIO": [ARROW_FILE_IO],
+    "org.apache.iceberg.gcp.gcs.GCSFileIO": [ARROW_FILE_IO],
+    "org.apache.iceberg.hadoop.HadoopFileIO": [ARROW_FILE_IO],
+    "org.apache.iceberg.aliyun.oss.OSSFileIO": [ARROW_FILE_IO],
+    "org.apache.iceberg.io.ResolvingFileIO": [ARROW_FILE_IO],
+    "org.apache.iceberg.aws.s3.S3FileIO": [ARROW_FILE_IO],
```

Comment:
I'm not sure whether it is worth it, but would it make sense to take an approach similar to [catalog loading](https://github.com/apache/iceberg/pull/5488/files#diff-a9e2f94cd7e8c7094195f92e4215b57e25a6ab1e30fd7a3264d4a24a05fc012fR51-R63)?

The map from scheme to implementation could use methods that load a known library using a function like in the other PR. Then the `_import_file_io` with a raw implementation class would only need to be used for overrides.

Reply:
The `load_` methods were created for loading the catalog in a lazy manner. We could create a method like `pyiceberg.io.pyarrow:load_file_io`, but I don't see what benefit that brings, at the cost of additional complexity.

Something like:
```
# Default to PyArrow
from pyiceberg.io.pyarrow import load_pyarrow()
return load_pyarrow()
``` 

But then we still need to have a fallback for FileIOs that are outside of the scope of the `pyiceberg` package. I like the current implementation because it very specifically imports the FileIO that we're looking for, and we don't import the class on the forehand, removing the possibility of importing optional dependencies.